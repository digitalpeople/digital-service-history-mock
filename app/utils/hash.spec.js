import bcrypt from 'bcrypt';
import {
  hashPassword,
  comparePassword,
} from './password';

jest.mock('bcrypt', () => ({
  hash: jest.fn(),
  compare: jest.fn(),
}));

describe('hashPassword', () => {
  it('should call bcrypt.hash with given parameters', () => {
    expect.assertions(1);

    const password = 'mock-password';

    hashPassword(password);

    expect(bcrypt.hash.mock.calls[0][0]).toMatchSnapshot();
  });
});

describe('comparePassword', () => {
  it('should call bcrypt.compare with given parameters', () => {
    expect.assertions(1);

    const password = 'mock-password';
    const targetPassword = 'stored-password';

    comparePassword(password), targetPassword;

    expect(bcrypt.compare.mock.calls[0][0]).toMatchSnapshot();
  });
});
