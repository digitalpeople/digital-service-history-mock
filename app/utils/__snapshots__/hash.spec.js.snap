// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comparePassword should call bcrypt.compare with given parameters 1`] = `"mock-password"`;

exports[`hashPassword should call bcrypt.hash with given parameters 1`] = `"mock-password"`;
